### DISCUSSION TITLE
Issue with Terraform with Snowflake: '"private_key": conflicts with private_key_path'

### DISCUSSION THREAD

**Discussion by VinP-PA** [Snowflake-Labs/terraform-provider-snowflake#1943] [Q&A]
I am having some issues with trying to deploy a simple Snowflake database and warehouse via Github Actions where I keep getting the following error: 

```
 Error: Conflicting configuration arguments
   with provider["registry.terraform.io/snowflake-labs/snowflake"],
   on provider.tf line 12, in provider "snowflake":
   12: provider "snowflake" 
 "private_key": conflicts with private_key_path || "private_key_path": conflicts with private_key

```
I have tried the following with no luck:

**Attempt#1(With environment variables)**
.github/workflows/pipeline.yml
            
      - name: Setup authentication
        run: |
          export SNOWFLAKE_USER="${{env.SNOWFLAKE_USER}}"
          export SNOWFLAKE_ACCOUNT="${{env.SNOWFLAKE_ACCOUNT}}"
          export SNOWFLAKE_REGION="${{env.SNOWFLAKE_REGION}}"
          export SNOWFLAKE_PRIVATE_KEY_PATH="${{env.SNOWFLAKE_PRIVATE_KEY_PATH}}"

      - name: Setup Terraform v1.4.2
        uses: hashicorp/setup-Terraform@v2
        with:
          terraform_version: 1.4.2
          terraform_wrapper: false

      - name: Setup Terraform version
        run: terraform --version

      - name: Terraform init
        working-directory: ./terraform
        run: terraform init

      - name: Terraform Validate
        working-directory: ./terraform
        run: terraform validate

provider.tf

    terraform {
      required_providers {
        snowflake = {
          source  = "Snowflake-Labs/snowflake"
          version = "~> 0.67"
        }
      }
    
      required_version = ">= 1.2.0"
    }
    
    provider "snowflake" {
      role  = "SYSADMIN"
    }

**Attempt#2-4(Without environment variables, authentication in Provider)**

provider.tf

    terraform {
      required_providers {
        snowflake = {
          source  = "Snowflake-Labs/snowflake"
          version = "~> 0.67"
        }
      }
    
      required_version = ">= 1.2.0"
    }
    
    provider "snowflake" {
      username = "tf-snow"
      account = "AB12345"
      region = "eu-west-2"
      role  = "SYSADMIN"
      password = "Pa$$W0rd!" # Attempt 2
      private_key = var.snowflake_private_key # Attempt 3
      private_key_path = "./snowflake_tf_key.p8" # Attempt 4
    }

**Attempt#5(With .snowflake/config file)**

    [default]
    account = "AB12345"
    username = "tf-snow"
    password='Pa$$W0rd!'
    role='SYSADMIN'

For reference:

main.tf

    resource "snowflake_database" "database" {
      name     = "TF_DEMO"
    }
    
    resource "snowflake_warehouse" "warehouse" {
      name           = "TF_DEMO"
      warehouse_size = "small"
      auto_suspend = 60
    }

I've run the terraform scripts via the command line using the environment variables and it has worked successfully.

Any help around this would be greatly appreciated.

**Comment by mikehawkins1** [comment]
Regarding the error, private_key and private_key_path conflicting due to the private_key is giving the provider block the key directly, and private_key_path i giving the provider block the path to the key. You need to do one or the other and not both. I would recommend giving it the path to the key. 
your provider block should look like this.
```
 provider "snowflake" { 
  username = "tf-snow"  
  account = "AB12345"  
  region = "eu-west-2"
  role  = "SYSADMIN"
  password = "Pa$$W0rd!"
  private_key_path = "./snowflake_tf_key.p8" # Attempt 4
}
```
